sdk/core/core/src/az_span_private.h

```diff
@@ -35,6 +35,59 @@ _az_span_scan_until(az_span self, _az_predicate predicate, int32_t* out_index);
 
 AZ_NODISCARD az_result _az_is_expected_span(az_span* self, az_span expected);
 
+/**
+ * @brief Removes any white space from right and left of \p span. Function will create a new
```

Comment:
nit: I am a bit biased, but I like the wording of docs like these:
https://docs.microsoft.com/en-us/dotnet/api/system.memoryextensions.trim?view=netcore-3.1#System_MemoryExtensions_Trim_System_Span_System_Char__

```suggestion
 * @brief Removes all leading and trailing white space characters from the \p span. Function will create a new
```

Also, consider moving the second sentence into `@note` instead to keep the `@brief` short/single sentence.